LibreNMS/Alert/Transport/Ilert.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/* Copyright (C) 2014 Daniel Preussker <f0o@devilcode.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>. */
+
+/**
+ * iLert Transport
+ *
+ * @author t.plueer <t.plueer@first-colo.net>, l.prosch <l.prosch@first-colo.net>
+ * @copyright 2024 firstcolo
+ * @copyright 2022 f0o, LibreNMS
+ * @license GPL
+ */
+
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+use LibreNMS\Enum\AlertState;
+
+class Ilert extends Transport
+{
+    public function name(): string
+    {
+        return 'ilert';
+    }
+
+    public function deliverAlert(array $alert_data): bool
+    {
+        if ($alert_data['state'] == AlertState::RECOVERED) {
+            $alert_data['event_type'] = 'RESOLVE';
+        } elseif ($alert_data['state'] == AlertState::ACKNOWLEDGED) {
+            $alert_data['event_type'] = 'ACCEPT';
+        } else {
+            $alert_data['event_type'] = 'ALERT';
+        }
+        $opts = '';
+        if (! empty($this->config)) {
+            $opts = $this->config['api-token'];
+        }
+
+        return $this->contactILert($alert_data, $opts);
+    }
+
+    public function contactILert($alert_data, $opts): bool
+    {
+        $data = ['apiKey' => $opts, 'eventType' => $alert_data['event_type'], 'summary' => $alert_data['msg']];
+        $data = json_encode($data);
+
+        $curl = curl_init('https://api.ilert.com/api/events');
```

Comment:
Please use the http class instead, https://github.com/librenms/librenms/blob/master/LibreNMS/Util/Http.php

The documention for it is here; https://laravel.com/docs/9.x/http-client

Reply:
Please use the http class instead, https://github.com/librenms/librenms/blob/master/LibreNMS/Util/Http.php

The documention for it is here; https://laravel.com/docs/9.x/http-client